### ISSUE TITLE
is_subclass_of & InvalidStringClass

### ISSUE THREAD

**Issue by SignpostMarv** [vimeo/psalm#1172]
`is_subclass_of();` does not flag a string as being usable as a class when calling public static methods:

* https://getpsalm.org/r/dbd259f27b

```php
<?php
interface Foo {
  public static function Bar() : bool;
};

class FooClass implements Foo {
  public static function Bar() : bool {
    return true;
  }
}

  
function foo(string $className) : bool {
  if (is_subclass_of($className, Foo::class, true)) {
    return $className::Bar();
  }
  
  return false;
}
```

**Comment by muglug** [comment]
Thanks for ticketing. This `is_subclass_of` is now treated like `is_a` (with the last arg default changed)
